python/xorq/ibis_yaml/compiler.py

```diff
@@ -0,0 +1,265 @@
+import json
+import pathlib
+from pathlib import Path
+from typing import Any, Dict
+
+import dask
+import yaml
+
+import xorq as xo
+import xorq.common.utils.logging_utils as lu
+import xorq.vendor.ibis.expr.types as ir
+from xorq.common.utils.graph_utils import find_all_sources
+from xorq.ibis_yaml.common import SchemaRegistry, TranslationContext
+from xorq.ibis_yaml.config import config
+from xorq.ibis_yaml.sql import generate_sql_plans
+from xorq.ibis_yaml.translate import (
+    translate_from_yaml,
+    translate_to_yaml,
+)
+from xorq.ibis_yaml.utils import freeze
+from xorq.vendor.ibis.backends import Profile
+from xorq.vendor.ibis.common.collections import FrozenOrderedDict
+
+
+class CleanDictYAMLDumper(yaml.SafeDumper):
+    def represent_frozenordereddict(self, data):
+        return self.represent_dict(dict(data))
+
+    def ignore_aliases(self, data):
+        return True
```

Comment:
what is the purpose of `ignore_aliases`?

Reply:
YAML by default will create anchors/aliases for objects that repeat. I thought it was less readable so I disabled it. https://stackoverflow.com/questions/58091449/disabling-alias-for-yaml-file-in-python